### ISSUE TITLE
Question about custom import syntax

### ISSUE THREAD

**Issue by issammani** [webpack/webpack#16849]
# Bug report

Is it possible to support custom import syntax say `const {foo} = custom.import(module)` using webpack out of the box ? I looked at the module section of the docs but couldn't find anything. Babel is probably best suited for this, but if webpack supports this it would be one less dependency to think about 😄 

Thanks !

**What is the current behavior?**

**If the current behavior is a bug, please provide the steps to reproduce.**

**What is the expected behavior?**

**Other relevant information:**
webpack version:  
Node.js version: 
Operating System: 
Additional tools:

**Comment by TheLarkInn** [comment]
Any import syntax today is built of plugins in webpack's source code. Example: 

https://github.com/webpack/webpack/blob/main/lib/dependencies/HarmonyImportDependency.js

tl;dr yes you could implement your own syntax but it is not recommended because of complexity and somewhat reliance on less stable webpack internal APIs/hooks.
